lib/JiraClient.ts

```diff
@@ -38,30 +38,37 @@ export class JiraClient {
   }
 
   public async findTransition(issueId: string, transitionName: string): Promise<any> {
-    const transitions = (await this.sendJiraGet(`issue/${issueId}/transitions`))?.transitions ?? [];
+    const transitions = (await this.sendRestGet(`issue/${issueId}/transitions`))?.transitions ?? [];
     return transitions.find((x: any) => x.name === transitionName);
   }
 
-  public async transitionIssue(issueId: string, transition: any, fields:any = null): Promise<void> {
+  public async transitionIssue(issueId: string, transition: any, fields: any = null): Promise<void> {
     console.log(`${issueId}: Executing '${transition.name}' (${transition.id}) transition`);
-    await this.sendJiraPost(`issue/${issueId}/transitions`, { transition: { id: transition.id }, fields });
+    await this.sendRestPost(`issue/${issueId}/transitions`, { transition: { id: transition.id }, fields });
   }
 
-  public async assignIssue(issueId: string, userEmail: string): Promise<void> {
+  public async assignIssueToEmail(issueId: string, userEmail: string): Promise<void> {
     const accountId = await this.findAccountId(userEmail);
     if (accountId != null) {
-      console.log(`${issueId}: Assigning to ${accountId}`);
-      await this.sendJiraPut(`issue/${issueId}/assignee`, { accountId });
+      await this.assignIssueToAccount(issueId, accountId);
     }
   }
 
-  private async findAccountId(email: string): Promise<string> {
+  public async assignIssueToAccount(issueId: string, accountId: string): Promise<void> {
+    console.log(`${issueId}: Assigning to ${accountId}`);
+    await this.sendRestPut(`issue/${issueId}/assignee`, { accountId });
+  }
+
+  public async findAccountId(email: string): Promise<string> {
+    if (email == null) {
+      console.log('Could not find accountId, email is null');
+      return null;
+    }
```

Comment:
This null-check is new. Previously, this was called only after email was checked for null. Now, it's called every time.

The test org doesn't have SAML. And it applies also to bots.